### DISCUSSION TITLE
Top five things you wish you'd known when you adopted cFS

### DISCUSSION THREAD

**Discussion by spacesteps-bill** [nasa/cFS#407] [General]
Hi all,

I'm writing an articles series and creating a YouTube video talking about cFS. It's part of a small website I've started to help educate people on making satellites: https://spacesteps.com (pre-alpha site; check the Learn page for example articles)

I'm looking for things you'd want high school students and professionals alike to know about cFS. What pitfalls can they avoid? What knowledge and information would help them be successful?

And if you have any words of encouragement, that would be good, too. E.g. "It's a lot to understand at first, but it's better than..." or "Hey, check out this resource at...."

For what it's worth, I'm not monetizing this and it's meant to be a public service. (YouTube will still run ads, but I don't get any of that money unless 10,000+ people watch it...)

All the best,
-Bill

P.S. I plan to try to do this for F' and other architectures as well. Trying to just provide unbiased info.

**Comment by the-other-james** [comment]
I think the biggest things for new users are:
1) It is easy to get cFS running on your computer: cFS builds and runs on Linux by default
2) The [cFS repo](https://github.com/nasa/cFS) is the best way to get a cFS project going with matched versions for cFE/OSAL/PSP/example apps
3) Learning to use the ground station software is worth the time. Cosmos is fairly easy to get started with, and [OpenSatKit](https://github.com/OpenSatKit/OpenSatKit) is a great example of how to use it with cFS

The biggest pitfall in my opinion is finding the cFS version to use that has the right balance of new features and apps that have been updated for it. At this exact moment 6.7 seems to be the right balance, but as more of the apps are updated I think that will change to 7.0 (Caelum). For example, [CF](https://github.com/nasa/cf) has been updated, but [SCH](https://github.com/nasa/sch/) hasn't.

Finally, I think it's important to understand that cFS is still evolving. If you had asked this question a few years ago I would have had a longer list of pitfalls for you :)
